### PR TITLE
5.0: Update django.db.models.constraints.UniqueConstraint

### PR DESCRIPTION
**Pull request by steve148**
# I have made things!
    
Adds in nulls_distinct field to the UniqueConstraint class. Field can be None or a boolean.

https://docs.djangoproject.com/en/5.0/ref/models/constraints/#nulls-distinct

## Related issues

Refs #1493

## Feedback

None from me. Let me know if I missed anything.

### CODE REVIEW

**Review by sobolevn — COMMENTED** [review]
Please, reflect this line as well: https://github.com/django/django/blob/15fff62d5d8f28b6b61d63341535c9aafc64c865/django/db/models/constraints.py#L356

**Comment by steve148** [comment]
@sobolevn I amended the change, let me know if this is what you meant. I added the attribute to the class so it shows up as an arg on both the init and as an instance attribute. I also added a quick test to make sure it works.

File: tests/typecheck/db/models/test_constraints.yml [hunk 1]
```diff
@@ -35,3 +35,13 @@
         main:4: note: Possible overload variants:
         main:4: note:     .*
         main:4: note:     .*
+
+-   case: unique_constraint_nulls_distinct
```
Comment (sobolevn) [inline tests/typecheck/db/models/test_constraints.yml]: Test is not required, can you please remove it? They are really slow :(

Comment (steve148) [reply tests/typecheck/db/models/test_constraints.yml]: Works for me :+1:

**Review by sobolevn — APPROVED** [review]

**Comment by LaurierMantel** [comment]
This is big - thank you @steve148

**Pull request merged**